kenjutsu/blocks.py

```diff
@@ -12,7 +12,7 @@
 import kenjutsu.format
 
 
-def split_blocks(space_shape, block_shape, block_halo=None):
+def split_blocks(space_shape, block_shape, block_halo=None, index=None):
```

Comment:
We are only adding this flag to ease users into the new behavior, which is having indices always returned and no option for the old behavior. Given users can easily toss the index information aside if they don't want it, this shouldn't present a problem. The plan is as follows...

* 0.4.0 - Currently no flag.
  * Intended use -> `b, hb, thb = split_blocks(s, bs, bh)`
* 0.5.0 - Adding `index` flag that defaults to the old behavior with opt-in for new behavior.
  * Default value has the same behavior. So nothing breaks yet.
  * Issue a warning that the default value of this flag will change in the next minor release.
  * Issue a warning that this flag will be deprecated in the next minor release (pending removal in the following minor release).
  * Intended use -> `ib, b, hb, thb = split_blocks(s, bs, bh, index=True)`
* 0.6.0 - Change flag to default to `True`.
  * Drop warning about default behavior of this flag.
  * Change pending deprecation warning about removing this flag to a deprecation warning.
  * Intended use -> `ib, b, hb, thb = split_blocks(s, bs, bh)`
* 0.7.0 - Drop the flag.
  * Behavior is the same as if the flag were `True`.
  * No flag that anyone can set any more.
  * Deprecation warning about flag being dropped is removed.
  * Intended use -> `ib, b, hb, thb = split_blocks(s, bs, bh)` (same as 0.6.0)